### DISCUSSION TITLE
Is there an option to change the directory sorting when using "See all files.."

### DISCUSSION THREAD

**Discussion by mwnoo** [ArchiveBox/ArchiveBox#1602] [Q&A] [answered]
When I use the `See all files..` option on the snapshot index page, I would like to have the files sorted alphabetically. Is this possible?

Sometimes I add additional files to a snapshot folder with the year as a prefix (2020_* or 2023_*). But when hitting `See all files..` these files are not sorted by filename. Is this order fixed or is it possible to change this to sort by filename?

**Comment by pirate** [comment]
There is no option to change the sort currently, and I doubt it will be added soon as it's more of a power user escape hatch to see things that aren't in the normal UI.

I'd recommend pointing Nextcloud or another proper file management app at the archivebox data dir if you want a full fledged file manager web interface.

**Reply by mwnoo** [reply]
I don't need a full fledged file manager but the files listed in alphabetical order would be clearer I think. The order of the files listed on the `See all files..` page seems now random to me.

I found a way to order the DOM elements using this [script](https://alexwlchan.net/til/2024/sort-dom-elements-with-appendchild/):
```
const list = document.querySelector('ul');

[...list.children]
  .sort((a, b) => a.innerText > b.innerText ? 1 : -1)
  .forEach(node => list.appendChild(node));
```
But I need to run this script everytime I want to change the order of the files.

It's not clear to me where in the code the `See all files..` page is created or what template is used to generate this page.

**Reply by pirate** [reply, marked as answer]
> It's not clear to me where in the code the See all files.. page is created

It's generated automatically by Django's `django.views.static.directory_index`, it's not part of ArchiveBox's code, that's why it's not easy to add JS/modify the HTML without rewriting it entirely.

You can add that script you wrote to the page with something like https://violentmonkey.github.io/ though.

**Reply by mwnoo** [reply]
Works great with violentmonkey. Thanks for the suggestion!
